realm/realm-library/src/objectServer/java/io/realm/SyncSession.java

```diff
@@ -971,4 +978,5 @@ public void throwExceptionIfNeeded() {
     private static native byte nativeGetConnectionState(String localRealmPath);
     private static native void nativeStart(String localRealmPath);
     private static native void nativeStop(String localRealmPath);
+    private static native void nativeSetUrlPrefix(String localRealmPath, String urlPrefix);
```

Comment:
So it is safe to call this method at any point during the sessions lifecycle?

Reply:
We can call this method at any point, but it will only have an effect before the sync session is bound with an access token. Because the server always returns the same value this method is effectively idempotent.